src/components/mentoring/post/PostMentorSide.jsx

```diff
@@ -0,0 +1,240 @@
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import toast from "react-hot-toast";
+
+import { deletePostReq, donePostReq } from "../../../apis/mentoring/post";
+import {
+  acceptConnectionReq,
+  refuseConnectionReq,
+} from "../../../apis/mentoring/connetion";
+import { convertDateToAge } from "../../../utils/age";
+
+import Button from "../../common/Button";
+import FlagTag from "../../common/FlagTag";
+import Tag from "../../common/Tag";
+
+export default function PostMentorSide({ data }) {
+  const navigate = useNavigate();
+
+  const queryClient = useQueryClient();
+
+  const isMine = `${data.writer.uid}` === window.localStorage.getItem("uid");
+
+  const isDelete = !data.connections.some(
+    (connection) => connection.state !== "await"
+  );
+
+  const [checks, setchecks] = useState(
+    data.connections.reduce((acc, connection) => {
+      if (connection.state === "await")
+        return { ...acc, [connection.cid]: false };
+      else return acc;
+    }, {})
+  );
+
+  const handleCheckBoxChenage = (e) => {
+    const name = e.target.name;
+    if (name === "all") {
+      if (Object.keys(checks).every((val) => checks[val] === true))
```

Comment:
```suggestion
      if (e.target.checked)
```
이미 178번 라인에서 `checked={Object.values(checks).every((val) => val === true)}` 이렇게 checked 값을 업데이트 하고 있습니다.
따라서 checked 값을 참조만 해줘도 될 것 같아요.